src/api/users.ts

```diff
@@ -0,0 +1,55 @@
+import express from "express";
+import { PrismaClient } from "@prisma/client";
+const bcrypt = require("bcrypt")
+const jwt = require('jsonwebtoken');
+
+const SALT_ROUNDS = 10;
+
+const prisma = new PrismaClient();
+
+const {ACCESS_TOKEN_SECRET} = process.env;
+
+const usersRouter = express.Router();
+
+// GET /api/users
+usersRouter.get("/", async (req, res, next): Promise<void> => {
+    try {
+        const users = await prisma.user.findMany();
+        res.send({users});
+    } catch (e) {
+        next(e);
+    }
+})
+
+// POST /api/users/register
+usersRouter.post("/register", async (req, res, next) => {
+    try {
+        const { email, username, password } = req.body;
+        bcrypt.hash(password, SALT_ROUNDS, async function(err: Error | undefined, hash: string) {
+            const user = await prisma.user.create({
```

Comment:
The error is currently being swallowed because it's in a callback
```suggestion
        bcrypt.hash(password, SALT_ROUNDS, async function(err: Error | undefined, hash: string) {
            if (err) next(err);
            const user = await prisma.user.create({
```